File: 1328-break-a-palindrome/1328-break-a-palindrome.kt
class Solution {
    fun breakPalindrome(p: String): String {
        if (p.length == 1)
            return ""
        val q = p.toCharArray()
        for (i in p.indices)
        {
            if (q[i] != 'a'){
                q[i] = 'a'
                println(q)
                val v = q.joinToString ( "" )
                println(v)
                if (q.reversed().joinToString("") != q.joinToString(""))
                    return q.joinToString("")
            }
        }
        val x = p.toCharArray()
        x[p.length-1] = 'b'
        return x.joinToString("")
    }
}

File: 1328-break-a-palindrome/1328-break-a-palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
        //int n = palindrome.length();
        for(int i = 0; i<palindrome.length()/2; i++){
            if(palindrome[i] != 'a'){
                palindrome[i] = 'a';
                return palindrome;
            }
        }
        
        palindrome[palindrome.length()-1] = 'b';
        return palindrome.length()<2?"":palindrome;
    }
};